pkg/activator/handler/metric_handler.go

```diff
@@ -43,23 +40,19 @@ type MetricHandler struct {
 
 func (h *MetricHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	rev := RevisionFrom(r.Context())
-	reporterCtx, _ := metrics.PodRevisionContext(h.podName, activator.Name,
-		rev.Namespace, rev.Labels[serving.ServiceLabelKey], rev.Labels[serving.ConfigurationLabelKey], rev.Name)
 
-	start := time.Now()
+	serviceName := rev.Labels[serving.ServiceLabelKey]
+	configurationName := rev.Labels[serving.ConfigurationLabelKey]
 
-	rr := pkghttp.NewResponseRecorder(w, http.StatusOK)
-	defer func() {
-		err := recover()
-		latency := time.Since(start)
-		if err != nil {
-			reporterCtx := metrics.AugmentWithResponse(reporterCtx, http.StatusInternalServerError)
-			pkgmetrics.RecordBatch(reporterCtx, responseTimeInMsecM.M(float64(latency.Milliseconds())), requestCountM.M(1))
-			panic(err)
-		}
-		reporterCtx := metrics.AugmentWithResponse(reporterCtx, rr.ResponseCode)
```

Comment:
Maybe I'm missing something, but are we losing this response code with the new labeling/metrics reporting code?

Reply:
This is captured in the otelhttp library for both spans and metrics

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/45af9252727c122174f7adab90f6c24ab04f1a6b/instrumentation/net/http/otelhttp/handler.go#L188-L206

Reply:
So we can drop the custom code we had before

Reply:
That's nice!